src/flash.rs

```diff
@@ -34,3 +49,436 @@ impl ACR {
         unsafe { &(*FLASH::ptr()).acr }
     }
 }
+
+#[cfg(any(feature = "rm0433"))]
+/// BANK1 start address
+const BANK1_ADDR_START: usize = 0x0800_0000;
+#[cfg(any(feature = "rm0433"))]
+/// BANK1 end address
+const BANK1_ADDR_END: usize = 0x080F_FFFF;
+#[cfg(any(feature = "rm0433"))]
+/// BANK2 start address
+const BANK2_ADDR_START: usize = 0x0810_0000;
+#[cfg(any(feature = "rm0433"))]
+/// BANK2 end address
+const BANK2_ADDR_END: usize = 0x081F_FFFF;
+#[cfg(any(feature = "rm0433"))]
+/// Sector size of 128kb
+const SECTOR_SIZE: usize = 0x2_0000;
+
+#[cfg(any(feature = "rm0433"))]
+#[derive(Clone, Copy, PartialEq)]
+pub enum Bank {
+    /// User Bank 1
+    UserBank1,
+    /// User Bank 2
+    UserBank2,
+}
+
+#[cfg(any(feature = "rm0433"))]
+impl Bank {
+    /// The start address of the bank
+    fn start_address(&self) -> usize {
+        match *self {
+            Bank::UserBank1 => BANK1_ADDR_START,
+            Bank::UserBank2 => BANK2_ADDR_START,
+        }
+    }
+    /// The end address of the bank
+    fn end_address(&self) -> usize {
+        match *self {
+            Bank::UserBank1 => BANK1_ADDR_END,
+            Bank::UserBank2 => BANK2_ADDR_END,
+        }
+    }
+    /// The address of a given sector of a bank
+    fn sector_address(&self, sector: usize) -> usize {
+        if sector > self.num_sectors() {
+            panic!("Sector outside of bank range");
+        }
+        self.start_address() + (sector * SECTOR_SIZE)
+    }
+    /// The number of bytes within the bank
+    pub fn size(&self) -> usize {
+        match *self {
+            Bank::UserBank1 => (BANK1_ADDR_END - BANK1_ADDR_START),
+            Bank::UserBank2 => (BANK2_ADDR_END - BANK2_ADDR_START),
+        }
+    }
+    /// The number of sectors within the bank, including sector zero
+    pub fn num_sectors(&self) -> usize {
+        self.size() / SECTOR_SIZE
+    }
+    /// Sector size
+    pub fn sector_size(&self) -> usize {
+        match *self {
+            Bank::UserBank1 => SECTOR_SIZE,
+            Bank::UserBank2 => SECTOR_SIZE,
+        }
+    }
+}
+
+#[cfg(any(feature = "rm0433"))]
+#[derive(Copy, Clone, Debug)]
+/// Possible error states for flash operations.
+pub enum Error {
+    /// Error detected (by command execution, or because no command could be executed)
+    Illegal,
+    /// (Legal) command failed
+    Unlock,
+}
+
+#[cfg(any(feature = "rm0433"))]
+impl Flash {
+    fn bank_registers(&self, bank: Bank) -> &BANK {
+        match bank {
+            Bank::UserBank1 => self.regs.bank1(),
+            Bank::UserBank2 => self.regs.bank2(),
+        }
+    }
+
+    fn _configure_clocks(&mut self) {
+        // TODO: Configure clocks for optimal performance for both read and write operations
+
```

Comment:
The RCC module already sets the ACR parameters, so this section isn't needed

Reply:
Ah thanks @richardeoin I wasn't aware of this. I'll remove this